application_sdk/application/__init__.py

```diff
@@ -208,3 +217,55 @@ async def start_server(self):
             raise ValueError("Application server not initialized")
 
         await self.server.start()
+    
+    async def setup_mcp_server(self, mcp_name: Optional[str] = None):
```

Comment:
We should just have the `setup_server` and `start_server` methods in the `BaseApplication`, if the enable_mcp flag is passed as true to the `BaseApplication` it should setup the MCP server and mount it on the underlying FastAPI app. 

guide for the same: https://gofastmcp.com/integrations/fastapi#mounting-an-mcp-server